lib/brainstem/dsl/association.rb

```diff
@@ -18,8 +18,19 @@ def method_name
         end
       end
 
-      def run_on(model, helper_instance = Object.new)
-        options[:dynamic] ? helper_instance.instance_exec(model, &options[:dynamic]) : model.send(method_name)
+      def run_on(model, context, helper_instance = Object.new)
+        if options[:dynamic] && (!options[:lookup] || context[:models].size == 1)
+          proc = options[:dynamic]
+          if proc.arity > 0
```

Comment:
Minor: would this be more explicit if it was `proc.arity == 1`? Since it looks like this only ever supports one arg?